cli/azd/pkg/azsdk/storage/storage_blob_client.go

```diff
@@ -0,0 +1,208 @@
+	blobs := []*Blob{}
+
+	pager := bc.client.NewListBlobsFlatPager(bc.config.ContainerName, nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			err = fmt.Errorf("failed to get next page of blobs, %w", err)
+			var responseErr *azcore.ResponseError
+			if errors.As(err, &responseErr) {
+				return nil, fmt.Errorf("%w, %w", err, ErrContainerNotFound)
+			}
+			return nil, err
+		}
+
+		for _, blob := range page.Segment.BlobItems {
+			blobs = append(blobs, &Blob{
+				Name:         filepath.Base(*blob.Name),
+				Path:         *blob.Name,
+				CreationTime: *blob.Properties.CreationTime,
+				LastModified: *blob.Properties.LastModified,
+			})
+		}
+	}
+
+	return blobs, nil
+}
+
+// Download downloads a blob from the configured storage account container.
+func (bc *blobClient) Download(ctx context.Context, blobPath string) (io.ReadCloser, error) {
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return nil, err
+	}
+
+	resp, err := bc.client.DownloadStream(ctx, bc.config.ContainerName, blobPath, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download blob '%s', %w", blobPath, err)
+	}
+
+	return resp.Body, nil
+}
+
+// Upload uploads a blob to the configured storage account container.
+func (bc *blobClient) Upload(ctx context.Context, blobPath string, reader io.Reader) error {
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return err
+	}
+
+	_, err := bc.client.UploadStream(ctx, bc.config.ContainerName, blobPath, reader, nil)
+	if err != nil {
+		return fmt.Errorf("failed to upload blob '%s', %w", blobPath, err)
+	}
+
+	return nil
+}
+
+// Delete deletes a blob from the configured storage account container.
+func (bc *blobClient) Delete(ctx context.Context, blobPath string) error {
+	if err := bc.ensureContainerExists(ctx); err != nil {
+		return err
+	}
+
+	_, err := bc.client.DeleteBlob(ctx, bc.config.ContainerName, blobPath, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete blob '%s', %w", blobPath, err)
+	}
+
+	return nil
+}
+
+// Check if the specified container exists
+// If it doesn't already exist then create it
+func (bc *blobClient) ensureContainerExists(ctx context.Context) error {
+	exists := false
+
+	pager := bc.client.NewListContainersPager(nil)
+	for pager.More() {
+		if exists {
+			break
+		}
+
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed getting next page of containers: %w", err)
+		}
+
+		for _, container := range page.ContainerItems {
+			if *container.Name == bc.config.ContainerName {
+				exists = true
+				break
+			}
+		}
+	}
+
+	if !exists {
+		_, err := bc.client.CreateContainer(ctx, bc.config.ContainerName, nil)
+		if err != nil {
+			return fmt.Errorf("failed to create container '%s', %w", bc.config.ContainerName, err)
+		}
+	}
+
+	return nil
+}
+
+// createClient creates a new blob client and caches it for future use
+func NewBlobSdkClient(
+	ctx context.Context,
+	authManager *auth.Manager,
```

Comment:
Could we just take a TokenCredential here and have the caller supply it or does that not work great for our DI strategy?  It seems preferable to me to take the more specific abstraction if we can.

Reply:
If we go down the route of using TokenCredential here and have the client injected via IoC then I'm ok with that as long as we can use the authManager to create the token credential by default.